addons/product/models/product_pricelist_item.py

```diff
@@ -460,3 +460,9 @@ def _compute_price_before_discount(self, *args, **kwargs):
             pricelist_rule = pricelist_item
 
         return pricelist_rule._compute_base_price(*args, **kwargs)
+
+    @api.constrains('base_pricelist_id')
+    def _check_websites_in_company(self):
+        for record in self:
+            if record.base_pricelist_id.company_id != record.pricelist_id.company_id:
```

Comment:
While it's true that if there is a pricelist on the base_pricelist, then the new one (based
 on it), must have the same one, the opposite is false.
If the base pricelist work with every company, why couldn't the new pricelist be restricted to a company?